gaeguli/pipeline.c

```diff
@@ -673,6 +761,27 @@ gaeguli_pipeline_create_snapshot (GaeguliPipeline * self, GstBuffer * buffer)
   gst_buffer_unmap (buffer, &info);
 }
 
+static void
+gaeguli_pipeline_create_device_monitor (GaeguliPipeline * self)
+{
+  GstBus *bus;
+  GstCaps *caps;
+
+  self->device_monitor = gst_device_monitor_new ();
```

Comment:
This will create a new monitor each time vsrc pipeline is built and may leak the old one.

Maybe `device_monitor` doesn't even have to be a member of `GaeguliPipeline` struct. Just create the monitor locally, find the node ID and dispose it afterwards.